activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb

```diff
@@ -378,15 +378,15 @@ def bulk_change_table(table_name, operations)
               non_combinable_operations << procs if procs.present?
             else
               execute "ALTER TABLE #{quote_table_name(table_name)} #{sql_fragments.join(", ")}" unless sql_fragments.empty?
```

Comment:
Looks like this conditional is missing a test too, because it has a similar problem: if we've previously called a `_for_alter` method that returned only a proc, `sql_fragments` won't be `empty?`... it'll be `[[]]`.

Instead of the flatten below, I think we want `.concat` on 377 / 378.

Reply:
Right now, there is no case which would return only a proc. I believe the only way to recreate this in a test would be to only update the comment. However, [`change_column_for_alter`](https://github.com/rails/rails/blob/master/activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb#L699-L705) adds a line of sql (which would do nothing if only updating comment). I'm not sure if it's possible to remove this line: even if there were no options provided, there might be a type change. I guess we could change `change_column_sql` to check for those two possibilities?

Reply:
Ah okay, let's skip testing that for now, and we can revisit after this is pulled up to abstract (and we have a wider variety of callees to choose from)